react/features/base/ui/components/web/DialogWithTabs.tsx

```diff
@@ -0,0 +1,332 @@
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'space-between',
+
+            '@media (max-width: 607px)': {
+                padding: '16px 24px'
+            }
+        },
+
+        title: {
+            ...withPixelLineHeight(theme.typography.heading5),
+            color: `${theme.palette.text01} !important`,
+            margin: 0,
+            padding: 0
+        },
+
+        contentContainer: {
+            position: 'relative',
+            display: 'flex',
+            padding: '24px',
+            flexDirection: 'column',
+            overflow: 'hidden',
+            width: '100%',
+
+            '@media (max-width: 607px)': {
+                padding: '0'
+            }
+        },
+
+        buttonContainer: {
+            width: '100%',
+            boxSizing: 'border-box',
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'flex-end',
+            flexGrow: 0,
+
+            '@media (max-width: 607px)': {
+                justifyContent: 'space-between',
+                padding: '16px 24px'
+            }
+        },
+
+        backContainer: {
+            display: 'flex',
+            alignItems: 'center',
+
+            '& > button': {
+                marginRight: '24px'
+            }
+        },
+
+        closeIcon: {
+            '&:focus': {
+                boxShadow: 'none'
+            }
+        },
+
+        content: {
+            flexGrow: 1,
+            overflowY: 'auto',
+            width: '100%',
+            boxSizing: 'border-box',
+
+            '@media (max-width: 607px)': {
+                padding: '0 24px'
+            }
+        },
+
+        footer: {
+            justifyContent: 'flex-end',
+
+            '& button:last-child': {
+                marginLeft: '16px'
+            }
+        }
+    };
+});
+
+interface IObject {
+    [key: string]: string | string[] | boolean | number | number[] | {} | undefined;
+}
+
+export interface IDialogTab {
+    className?: string;
+    component: ComponentType<any>;
+    icon: Function;
+    labelKey: string;
+    name: string;
+    props?: IObject;
+    propsUpdateFunction?: (tabState: IObject, newProps: IObject) => IObject;
+    submit?: Function;
+}
+
+interface IProps extends IBaseProps {
+    defaultTab?: string;
+    tabs: IDialogTab[];
+}
+
+const DialogWithTabs = ({
+    className,
+    defaultTab,
+    titleKey,
+    tabs
+}: IProps) => {
+    const { classes, cx } = useStyles();
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const [ selectedTab, setSelectedTab ] = useState<string | undefined>(defaultTab ?? tabs[0].name);
+    const [ tabStates, setTabStates ] = useState(tabs.map(tab => tab.props));
+    const clientWidth = useSelector((state: IReduxState) => state['features/base/responsive-ui'].clientWidth);
+    const [ isMobile, setIsMobile ] = useState(false);
+
+    useEffect(() => {
+        if (clientWidth < 608) {
```

Comment:
this looks very arbitrary. If we do intend to use this value as a discriminator between mobile and anything larger, we should expose it in a commonly accessible place

Reply:
I'll extract it in a const, but considering it's only used in this component I think it should only be available here.